### ISSUE TITLE
Change control of standards documents

### ISSUE THREAD

**Issue by jogu** [ConsumerDataStandardsAustralia/standards-maintenance#15]
# Request For Clarification

The changes from 0.9.6 -> 1.0.0 seem to have been made in one or two large changes:

https://github.com/ConsumerDataStandardsAustralia/standards/commit/f22d5d2c720c13ccd83275cad9e4b56eec4a9043
https://github.com/ConsumerDataStandardsAustralia/standards/commit/5e3a8f80db2952424dc3eb3668cc1a5889db3cf6
(and possibly some other commits as well, the git history is a little hard to follow, I think there's possibly a "rebuild" process that's not yet been automated which makes it noisier?)

I can't find any description that covers why each change was made, any reasoning, security analysis, pro/cons, etc. Is this available please?

I understand that 1.0.0 is ready for vendors to implement, so can we assume that going forward each change to the standard will be made as a result of an issue in this issue tracker, with a pull request made linked to that issue and hence the final standard being traceable (via 'git blame') back to a description of the intention / rational / analysis etc? (I see there is mention of something like this on https://github.com/ConsumerDataStandardsAustralia/standards-maintenance but it's not clear, e.g. will the discussions on the teleconferences will be minuted back to the relevant issue on the issue tracker? will comments received via email to csiro or otherwise made off-github be added to the relevant GitHub issue if they are to be taken into account?).

I can't find the documentation on the detail of the process, is this available somewhere? I guess it will be something like a branch will be opened for containing the 'draft' for each potential new spec version, then pull requests will be raised to that branch for each issue, some time given for members to review the contents of the pull request, then the pull request merged to the branch and the draft branch merged to 'master' branch when the cycle for that release completes?

Or if a process like this the above is not being followed, what is the model for public (or at least vendor etc) scrutiny of each change to the standards prior to a new version being published, and how is all the information about rational etc recorded & made available incase a decision later needs to be reconsidered?

**Comment by jogu** [comment]
I note the publication of https://github.com/ConsumerDataStandardsAustralia/standards/issues/50#issuecomment-541394618 and observe that it doesn't answer some of the above questions; in particular there is no description of how the actual text in the standard is changed and whether or not this would follow the expected model of having one pull request created that contains changes to the standard for one decision, linked to the issue and with the opportunity for the actual new text in the standard to be reviewed and potentially clarified/improved before being submitted for final approval.

**Comment by CDR-API-Stream** [comment]
As the approval process for the standards is via the Data Standards Chair the model of using decision proposals for changes has been adopted.

Following the approval of a change a branch and pull request is created but this is managed using a private staging repository.

Once the DSB team has reviewed the changes and confirmed that they align with the approved Decision the PR is merged and then the changes are pushed to the public repository.

As a result, while the public repository does not contain the PR and Branch used to create the changes, the commit history still contains all of the changes and can be used for comparison purposes.

**Comment by jogu** [comment]
Thanks for the response!

> Following the approval of a change a branch and pull request is created but this is managed using a private staging repository.
>
> Once the DSB team has reviewed the changes and confirmed that they align with the approved Decision the PR is merged and then the changes are pushed to the public repository.

I'm not sure I understand the rational for:

1. Why the review of whether the actual text changes reflects the agreed decision is done being in private

2. Why there is no public opportunity for review of the text prior to it being finalised

Whilst I understand that the legal need for review by the Data Standards Chair causes some deviation from the processes you'd expect open standards to follow (e.g. the processes used to agree the FAPI, OpenID Connect and OAuth 2 specs that the Australian standards built on top of), these deviations don't seem to relate to that requirement. Can you expand on the rational for te above lack of transparency please?

**Comment by CDR-API-Stream** [comment]
The only activity that is performed internally is the iterative development of the changes.  The consultation process and resulting output are all fully transparent and are open to critique.  There have been a number of occasions where the text incorporated to adopt a decision was ambiguous or in error and needed to be corrected and this was highlighted by the community.  These changes are also transparent.

The actually process for developing the standards artefacts is a choice for the DSB team.

It should be noted that the standards, while openly and transparently consulted upon, are not openly developed.  They are explicitly a centrally endorsed and managed set of artefacts.

**Comment by nghamilton** [comment]
@CDR-API-Stream Perhaps @jogu's point could be addressed by putting in a summary of changes into the description of a release, and tagging all our releases in Git? The details would then be visible in: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/releases
